test/inputs-redis-increments-test.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+const support = require('./support');
+const redis = require('../lib/redis');
+const RedisIncrements = require('../lib/inputs/redis-increments');
+
+describe('redis-increments', () => {
+
+  beforeEach(() => process.env.REDIS_HOST = 'redis://127.0.0.1:6379');
+
+  it('recognizes download and impression records', () => {
+    let incr = new RedisIncrements();
+    expect(incr.check({})).to.be.false;
+    expect(incr.check({type: 'foobar', feederPodcast: 1})).to.be.false;
+    expect(incr.check({type: 'download', feederPodcast: 1})).to.be.true;
+    expect(incr.check({type: 'impression', feederPodcast: 1})).to.be.true;
+    expect(incr.check({type: 'download', feederPodcast: 1, isDuplicate: true})).to.be.false;
+    expect(incr.check({type: 'download', feederPodcast: 1, isDuplicate: false})).to.be.true;
+  });
+
+  it('inserts nothing for no records', () => {
+    let incr = new RedisIncrements();
+    return incr.insert().then(result => {
+      expect(result.length).to.equal(0);
+    });
+  });
+
+  it('inserts nothing for records without feeder ids/guids', () => {
+    let incr = new RedisIncrements([
+      record(1490827132, 'download', null, null),
+      record(1490827132, 'impression', null, null)
+    ]);
+    return incr.insert().then(result => {
+      expect(result.length).to.equal(0);
+    });
+  });
+
+  it('inserts nothing for no redis host', () => {
+    process.env.REDIS_HOST = '';
+    let records = [record(1490827132, 'download', 1234, 'abcd')];
+    let incr = new RedisIncrements(records);
+    return incr.insert().then(result => {
+      expect(result.length).to.equal(0);
+
+      process.env.REDIS_HOST = 'whatev';
+      incr = new RedisIncrements(records);
+      return incr.insert();
+    }).then(result => {
+      expect(result.length).to.equal(1);
+    });
+  });
+
+  it('increments redis counts', () => {
+    let incr = new RedisIncrements([
+      record(1490827132, 'download', 1234, 'abcd'),
+      record(1490827132, 'download', 1234, 'efgh'),
+      record(1490828132, 'download', 1234, 'abcd'),
+      record(1490827132, 'impression', 1234, null),
+      record(1490827132, 'impression', 1234, 'efgh', true)
+    ]);
+    return incr.insert().then(result => {
+      expect(result.length).to.equal(1);
+      expect(result[0].dest).to.equal('redis://127.0.0.1');
+      expect(result[0].count).to.equal(4 * 5 + 3 * 5); // skips null episode impression
```

Comment:
ok, so we've got 
4 downloads/impressions for podcast `1234`, 
2 downloads for episode `abcd`, 
1 download for episode `efgh`, and 
1 dup download for episode `abcd` (skipped)

so we get 4 podcast things plus 3 episode things, and they're multiplied by the number of different time intervals we're recording info for in castle (15MIN etc). 

is all that correct? writing it out is helping me keep track in my head

Reply:
I was thinking of it as:

- [4 podcast-ids] * [5 keys per podcast]
- [3 episode-guids] * [5 keys per episode]

But yes, you've got it.